services/vm-watcher/vm_watcher_service_test.go

```diff
@@ -356,6 +356,64 @@ var _ = Describe(
 					})))
 				})
 
+				When("a Zone that is being deleted", func() {
+					BeforeEach(func() {
+						outerInitEnvFn := initEnvFn
+
+						initEnvFn = func(ctx *builder.IntegrationTestContextForVCSim) {
+							outerInitEnvFn(ctx)
+
+							zone := &topologyv1.Zone{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:      "bogus",
+									Namespace: vcSimCtx.NSInfo.Namespace,
+									Finalizers: []string{
+										"fake/finalizer", // Finalizer to prevent deletion
+									},
+								},
+								Spec: topologyv1.ZoneSpec{},
+							}
+							Expect(ctx.Client.Create(ctx, zone)).To(Succeed())
+							// Delete the Zone so the deletion timestamp is set.
+							Expect(ctx.Client.Delete(ctx, zone)).To(Succeed())
+						}
+					})
+
+					When("the zone has the zone finalizer", func() {
+						Specify("a reconcile request should be received", func() {
+							chanSource := cource.FromContext(ctx, "VirtualMachine")
+							var e event.GenericEvent
+							Eventually(chanSource).Should(Receive(&e, Equal(event.GenericEvent{
+								Object: &vmopv1.VirtualMachine{
+									ObjectMeta: metav1.ObjectMeta{
+										Namespace: vcSimCtx.NSInfo.Namespace,
+										Name:      vmName,
+									},
+								},
+							})))
+						})
+					})
+
+					When("the zone does not have the finalizer", func() {
```

Comment:
I think we should also clarify which "finalizer" here. If it's just a general one, then this is not the case since the zone has a fake finalizer added in parent's BeforeEach().